STYLE.md

```diff
@@ -0,0 +1,12 @@
+# Go coding style
+
+* The xDS Relay source code is formatted using
+  [golangci-lint](https://github.com/golangci/golangci-lint). Thus all white spaces, etc.
+  issues are taken care of automatically. Azure pipeline tests will automatically check the code
+  format and fail.
+* More detailed descriptions of each linter can be found
+  [here](https://github.com/golangci/golangci-lint/tree/v1.23.6#enabled-by-default-linters).
+
+# Recommended Go style guides
+
+* [Effective Go](https://golang.org/doc/effective_go.html)
```

Comment:
Not saying we need to figure this out in this PR, but we should add more here. Here are some style guides to get you inspired.

Here are the Kubernetes coding conventions: https://github.com/kubernetes/community/blob/master/contributors/guide/coding-conventions.md

Uber's Go Style Guide: https://github.com/uber-go/guide/blob/master/style.md

Google's Go Code Review Comments: https://github.com/golang/go/wiki/CodeReviewComments

Feel free to pick and choose or just adopt what you think is best.

Reply:
I'll add Go Code Review Comments. I think with Go in particular, linters do a pretty good job of catching style smells. I'm wary of adding too many custom style guides in case of contradiction with the others. We can add more as we iterate on this project.

Reply:
Ah yeah. I wasn't asking to put all of these in here. I think we should also have a section of our own and take what we want! :)